config/configgrpc/configgrpc.go

```diff
@@ -192,6 +192,29 @@ func (gcs *ClientConfig) ToClientConn(ctx context.Context, host component.Host,
 	return grpc.DialContext(ctx, gcs.SanitizedEndpoint(), opts...)
 }
 
+func (gcs *ClientConfig) toTransportCredentials(settings component.TelemetrySettings) (credentials.TransportCredentials, error) {
+	tlsSetting := &gcs.TLSSetting
+	if gcs.TLSSetting.Insecure {
+		// Per https://github.com/alanwest/opentelemetry-specification/blob/main/specification/protocol/exporter.md
+		// If the endpoint uses the https scheme, then the TLS insecure setting must be overridden.
+		if strings.HasPrefix(gcs.Endpoint, "https://") {
+			settings.Logger.Warn("This https endpoint overrides the TLS `insecure` setting. Set the `insecure` option to false or drop the `https` scheme to remove this warning.", zap.String("endpoint", gcs.Endpoint))
+			tlsSetting = &configtls.TLSClientSetting{}
+			*tlsSetting = gcs.TLSSetting
+			tlsSetting.Insecure = false
+		}
```

Comment:
As mentioned in a different PR, we should actually have a custom Unmarshal and override the `Insecure` based on the prefix and remove the prefix from the endpoint.

Reply:
ah, right. Sorry. I'll go do it that way instead.

Reply:
During unmarshalling you don't have a logger, and not sure it brings anything useful to the user. If you really want we can error.

Reply:
alright - how about a config converter? These have a logger.

Reply:
> alright - how about a config converter? These have a logger.

They probably don't as well, since they are applied before we know the final config.

Reply:
I checked - we use log.Println in our distro :( ok then